Allow unixtime-style timestamp extraction

rsommer opened the pull request:
Some backup solutions prefer using UNIX timestamps for tagging backup files or directories. In order to enable the rotation of such files/directories some extra handling is added to enable scanning for UNIX timestamps in file- or directory-names.

neopostmodern commented:
Are there plans to still merge this?

dfredell commented:
We would also be interested in this feature. @xolox Is this repo still maintained?

dfredell commented on README.rst:
```diff
@@ -155,8 +155,9 @@ intended you have no right to complain ;-).
    timestamps from filenames. ``PATTERN`` is expected to be a Python compatible
-   regular expression that must define the named capture groups 'year',
-   'month' and 'day' and may define 'hour', 'minute' and 'second'."
+   regular expression that must define a named capture group 'unixtime' or the
```
@rsommer can you provide better documentations or an example of how to use this new unixtime capture group? I had to read the test case to understand it. There is a section lower in the docs containing some Regex examples, [Supported configuration options](https://github.com/xolox/python-rotate-backups#id7). That would be a good spot for more elaboration.

rsommer replied:
I added a short example to the README file, hope that helps - otherwise this seems to be abandoned, since the original pull request was untouched for over a year :(

ernstki replied:
I, too, used to think "no commits in a year! oh no, it's _abandoned_," as if the author of an open-source project owed _me_, a random internet visitor, something, anything at all.

If it helps, think of the project as "finished" instead, or "[in hobby mode](https://tenfourfox.blogspot.com/2020/04/the-end-of-tenfourfox-and-what-ive.html)."

Yes, making the GitHub repo read-only or updating the README are helpful signals about the authors' intent to accept PRs, but authors of free, open source software are under no obligation to do that, either.

rsommer committed:
Add simple test for UNIX timestamp extraction

rsommer committed:
Allow unixtime-style timestamp extraction

Some backup solutions prefer using UNIX timestamps for tagging backup
files or directories. In order to enable the rotation of such
files/directories some extra handling is added to enable scanning for UNIX
timestamps in file- or directory-names.

rsommer committed:
Add a short example for UNIX timestamp extraction

aroberts commented:
I assume you've worked around this, after all this time, but I published a docker image ([here](https://hub.docker.com/repository/docker/aroberts/rotate-backups)) of my fork ([here](https://github.com/aroberts/python-rotate-backups)), which includes your PR. I am not planning to maintain anything in pypi as it's outside my use case, but hopefully this is useful to someone.

https://hub.docker.com/repository/docker/aroberts/rotate-backups
